odfuzz/generators.py

```diff
@@ -3,120 +3,223 @@
-        return '{}f'.format(round(random.uniform(1.18e-20, 3.40e+20), 7))
+    def generate(format='uri'):
+        value = '{}f'.format(round(random.uniform(1.18e-20, 3.40e+20), 7))
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
 
 
 class EdmGuid:
     @staticmethod
-    def generate():
-        return 'guid\'{0}\''.format(str(uuid.UUID(int=random.getrandbits(128), version=4)))
+    def generate(format='uri'):
+        value = 'guid\'{0}\''.format(str(uuid.UUID(int=random.getrandbits(128), version=4)))
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
 
 
 class EdmInt16:
     @staticmethod
-    def generate():
-        return str(random.randint(-32768, 32767))
+    def generate(format='uri'):
+        value = str(random.randint(-32768, 32767))
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
 
 
 class EdmInt32:
     @staticmethod
-    def generate():
-        return str(random.randint(-2147483648, 2147483647))
+    def generate(format='uri'):
+        value = str(random.randint(-2147483648, 2147483647))
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
 
 
 class EdmInt64:
     @staticmethod
-    def generate():
-        return str(random.randint(-9223372036854775808, 9223372036854775807)) + 'L'
+    def generate(format='uri'):
+        value = str(random.randint(-9223372036854775808, 9223372036854775807)) + 'L'
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
 
 
 class EdmSByte:
     @staticmethod
-    def generate():
-        return str(random.randint(-128, 127))
-
+    def generate(format='uri'):
+        value = str(random.randint(-128, 127))
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
 
 class EdmString:
     @staticmethod
-    def generate(self):
-        return '\'{}\''.format(RandomGenerator.random_string(self.max_length))
+    def generate(self,format='uri'):
+        value = '\'{}\''.format(RandomGenerator.random_string(self.max_length))
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
 
 
 class EdmTime:
     @staticmethod
-    def generate():
+    def generate(format='uri'):
```

Comment:
This is a good catch @lubosmj, thx. @AlapanGhosh, rename it to for example "usage_in" or "generator_format", or something similar to indicate the indentation of possible targets URI/POST body. You could also introduce enum class, so the generate() function would accept really just the values for example GeneratorFormat.URI and GeneratorFormat.POST_BODY and not basically any string. See https://docs.python.org/3/library/enum.html

It would be also nice to address lacks of test coverage mentioned by Lubos if possible, especially since the code is heavily modified now.

Reply:
@lubosmj That's a great point. Will do :)
@phanak-sap Yeah, better to use enum for the values. As for the tests, we might need to discuss this briefly once, maybe some time next week.

Reply:
yeah, we intentionally did not use enums in the DirectBuilder constructor, so at the momement it is more easy to tie together in the future with CLI usage and ENV_VARS parsing (no standard way for enum values). But here it is only internally facing code, so enum would be great option.